mlx/compile.cpp

```diff
@@ -82,7 +83,11 @@ Compiled::Compiled(
       inputs_(std::move(inputs)),
       outputs_(std::move(outputs)),
       tape_(std::move(tape)),
-      constant_ids_(std::move(constant_ids)) {}
+      constant_ids_(std::move(constant_ids)),
+      is_constant_([this](size_t i) {
+        return constant_ids_.find(inputs_[i].id()) != constant_ids_.end();
+      }),
+      kernel_lib_(build_lib_name(inputs_, outputs_, tape_, constant_ids_)) {}
```

Comment:
This is kind of a nit.. but maybe it makes sense to just move that function into the constructor?  Then we wouldn't need to include the backend header. And it's nice to have the code that this file uses centralized.

Reply:
Yeah it makes sense moving the code, I have updated the PR. The backend header is still needed though since utils like `print_constant` are used.